astropy/io/fits/util.py

```diff
@@ -824,12 +824,24 @@ def _array_to_file_like(arr, fileobj):
     `numpy.ndarray.tofile`).
     """
 
+    if len(arr) == 0:
```

Comment:
I think the code is fine as-is personally, I don't think ``if not len(arr)`` is as readable...